sphinx_gallery/gen_rst.py

```diff
@@ -604,58 +600,113 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
         sys.argv[0] = src_file
         sys.argv[1:] = []
 
-    for blabel, bcontent, lineno in script_blocks:
-        if blabel == 'code':
-            code_output, rtime = execute_code_block(compiler, src_file,
-                                                    bcontent, lineno,
-                                                    example_globals,
-                                                    block_vars, gallery_conf)
+    t_start = time()
+    output_blocks = [execute_code_block(compiler, src_file, block,
+                                        example_globals,
+                                        block_vars, gallery_conf)
+                     for block in script_blocks]
+    time_elapsed = time() - t_start
+
+    sys.argv = argv_orig
+    clean_modules()
+
+    example_rst = rst_notebook_cell(script_blocks, output_blocks,
+                                    file_conf, gallery_conf)
+    save_rst_notebook(example_rst, example_file.replace('.py', '.rst'),
+                      time_elapsed, gallery_conf)
+
+    save_thumbnail(image_path_template, src_file, file_conf, gallery_conf)
+
+    example_nb = jupyter_notebook(script_blocks)
+    save_notebook(example_nb, example_file.replace('.py', '.ipynb'))
+
+    if block_vars['execute_script']:
+        logger.debug("%s ran in : %.2g seconds\n", src_file, time_elapsed)
+        # Writes md5 checksum if example has build correctly
+        # not failed and was initially meant to run(no-plot shall not cache
```

Comment:
missing space between run and (no-plot. Bonus points if you improve the wording of the comment because it is a bit hard to follow (saying many things at once).